perlmod/Fink/PkgVersion.pm

```diff
@@ -1172,13 +1172,17 @@ sub get_script {
 			# path-prefix-*wraps gcc and g++, system-perl configure hardcodes
 			# gcc-4.x, which is not wrapped or necessarily even present.
 			my ($perldirectory, $perlarchdir, $perlcmd) = $self->get_perl_dir_arch();
-			my $archflags = 'ARCHFLAGS=""'; # prevent Apple's perl from building fat
+			my $sdkpath = `xcrun --sdk macosx --show-sdk-path`;
+			my $perlversion = substr($^V, 1, 4); # hack to extract '5.18'
+			# prevent Apple's perl from building fat
+			my $archflags = 'ARCHFLAGS="-I$sdkpath/System/Library/Perl/$perlversion/darwin-thread-multi-2level/CORE"';
 			$default_script =
 				"$archflags $perlcmd Makefile.PL \%c\n".
 				"/usr/bin/make CC=gcc CXX=g++\n";
 		} elsif ($type eq 'modulebuild') {
 			my ($perldirectory, $perlarchdir, $perlcmd) = $self->get_perl_dir_arch();
-			my $archflags = 'ARCHFLAGS=""'; # prevent Apple's perl from building fat
+			# prevent Apple's perl from building fat
+			my $archflags = 'ARCHFLAGS="-I$sdkpath/System/Library/Perl/$perlversion/darwin-thread-multi-2level/CORE"';
```

Comment:
sdkpath and perlversion won't be set here as it's set in the previous block and not outside of the if.

Reply:
Fixed. Indeed, much more was wrong there (missing `chomp`, bad quotes leading to var not being interpolated etc.) -- I now verified that this works by test building some perl modules.